### ISSUE TITLE
Support of differentiation of multi-arg calls in the forward mode

### ISSUE THREAD

**Issue by blide** [vgvassilev/clad#168]
```c++
double multiarg(double x, double y) {
  return x + y;
}
double f(double x) {
  return multiarg(x, 1);
}
clad::differentiate(f);
```
Now will fail with:
```
clang-9: /src/tools/clad/lib/Differentiator/DerivativeBuilder.cpp:1456: clad::StmtDiff
clad::ForwardModeVisitor::VisitCallExpr(const clang::CallExpr*): Assertion `(CE->getNumArgs() <= 1) &&
"forward differentiation of multi-arg calls is currently broken"' failed.
```

**Comment by sudo-panda** [comment]
@vgvassilev please close this issue too

**Comment by sudo-panda** [comment]
@efremale suggests in #237:

>By the way, it would be good to make a similar change of parameter structure for forward mode too at some point to fix #168
>
>The idea (not specific for arrays but works in the same for arrays) is:
>For `double f(double a, double b)`, any application of clad::differentiate on f will generate/use
>double `f_jvp(double a, double da, double b, double db)`, where
>`f_da(a, b)` is `f_jvp(a, 1, b, 0)` and
>`f_db(a, b)` is `f_jvp(a, 0, b, 1)` and in general
>for any `x`, `y`, `f_jvp(a, x, b, y)` is equal to matrix-vector product `matmul(f_jacobian(a, b), {x, y})`, i.e the Jacobian-vector product (JVP).
>In particular in nested function calls the value of `f_jvp(a, _d_a, b, _d_b)` is exactly what we need to fix the issue #168.
>
>Note that this does not require explicit evaluation of jacobian, forward mode naturally computes JVP faster than evaluating Jacobian and matmul on it.
>
>This does not require any change in forward mode algorithm code, only in function signature. Current code works in the same way, it only hardcodes `_d_a = 1`, `_d_b = 0` e.g. inside `f_d_a` rather than accepting general values of `_d_a`, `_d_b` (which will give JVP in the result).
